fluentui_core/src/main/java/com/microsoft/fluentui/compose/ModalPopup.kt

```diff
@@ -56,53 +67,70 @@ import java.util.UUID
     content: @Composable () -> Unit,
 ) {
+    val properties = PopupProperties()
     val view = LocalView.current
-    val id = rememberSaveable { UUID.randomUUID() }
+    val density = LocalDensity.current
+    val layoutDirection = LocalLayoutDirection.current
     val parentComposition = rememberCompositionContext()
     val currentContent by rememberUpdatedState(content)
-    val layoutDirection = LocalLayoutDirection.current
+    val id = rememberSaveable { UUID.randomUUID() }
     val modalWindow = remember {
         ModalWindow(
             onDismissRequest = onDismissRequest,
+            properties = properties,
             composeView = view,
+            density = density,
             saveId = id
         ).apply {
-            setCustomContent(
-                parent = parentComposition,
-                content = {
-                    Box(
-                        Modifier
-                            .semantics { this.popup() }
-                            // Get the size of the content
-                            .onSizeChanged {
-                                popupContentSize = it
-                            }
-                            // Hide the popup while we can't position it correctly
-                            .alpha(if (canCalculatePosition) 1f else 0f)
-                            .windowInsetsPadding(
-                                convertWindowInsetsCompatTypeToWindowInsets(windowInsetsType)
-                            )
-                            .imePadding()
-                    ) {
-                        currentContent()
-                    }
+            setCustomContent(parentComposition) {
+                Box(
+                    Modifier
+                        .semantics { this.popup() }
+
+                        .onSizeChanged {
+                            popupContentSize = it
+                        }
+
+                        .alpha(if (canCalculatePosition) 1f else 0f)
+                        .windowInsetsPadding(
+                            convertWindowInsetsCompatTypeToWindowInsets(windowInsetsType)
+                        )
+                        .imePadding()
+                ) {
+                    currentContent()
                 }
-            )
+            }
         }
     }
 
     DisposableEffect(modalWindow) {
         modalWindow.show()
-        modalWindow.superSetLayoutDirection(layoutDirection)
+        modalWindow.updateParameters(
+            onDismissRequest = onDismissRequest,
+            properties = properties,
+            layoutDirection = layoutDirection
+        )
         onDispose {
             modalWindow.disposeComposition()
             modalWindow.dismiss()
         }
     }
+
+    Layout(
+        content = {},
+        modifier = Modifier
+            .onGloballyPositioned { childCoordinates ->
+                val parentCoordinates = childCoordinates.parentLayoutCoordinates!!
```

Comment:
can we avoid using !! operator if possible?

Reply:
Updated